### ISSUE TITLE
Move to extended regex in sed broke the command on my machine

### ISSUE THREAD

**Issue by davised** [macdub/pyenv-multiuser#13]
**Describe the bug**

Looks like the fix in #12 seems to be broken on my end.

**To Reproduce**

As an example:

```console
$ sed -r 's/${PYENV_ROOT}\(["'\'']*\?\)\(\/shims\/\?\)\?/${PYENV_LOCAL_SHIM}\1/g' /local/software/pyenv-2.4.0/libexec/pyenv-init
sed: -e expression #1, char 67: Invalid preceding regular expression
```

**Expected behavior**

Removing the `-r` resolves the error, but I don't think the find/replace works properly after that.

According to the gnu sed manual, I think I was able to fix the regex below https://www.gnu.org/software/sed/manual/html_node/ERE-syntax.html

```console
sed -ri 's!\$\{PYENV_ROOT\}(["'\'']*?)(/shims/?)?!${PYENV_LOCAL_SHIM}\1!g' $filename
```
However, this makes some changes to the pyenv-init function that I think are unintended, as it makes the shims and versions dir look at the PYENV_LOCAL_SHIM path instead of the PYENV_ROOT path.

**Desktop (please complete the following information):**
I'm on Rocky 9. sed is version 4.8.

**Comment by macdub** [comment]
I'll take a look and try to get a fix out by next week.

**Comment by kiseeb** [comment]
The same here on Ubuntu 24.04. modifying that line fixed it.

**Comment by jeremyschulman** [comment]
I am hitting the same issue as @davised.  Would love to use this plugin!  Thank you for you time and efforts to fix this issue.

**Comment by macdub** [comment]
I've released v1.0.8 which should resolve this issue.
Thank you for your patience.
